src/ListTabulator/index.ts

```diff
@@ -366,353 +445,450 @@ export default class ListTabulator {
+    const itemEl = this.renderer!.renderItem(endingHTML, { checked: false });
 
     /**
-     * Restore the caret position
+     * Move new item after current
      */
-    this.caret.restore();
-  }
+    currentItem?.after(itemEl);
 
+    /**
+     * If current item has children, move them to the new item
+     */
+    if (itemChildren) {
+      itemEl.appendChild(itemChildren);
+    }
+
+    this.focusItem(itemEl);
+  }
 
   /**
-   * Reduce indentation for current item
-   *
-   * @param {KeyboardEvent} event - keydown
-   * @returns {void}
+   * Method that is used for merging current item with previous one
+   * Content of the current item would be appended to the previous item
+   * Current item children would not change nesting level
+   * @param currentItem - current item html element
    */
-  shiftTab(event: KeyboardEvent): void {
+  mergeCurrentItemWithPrevious(currentItem: HTMLElement): void {
+    const previousItem = currentItem.previousElementSibling;
+
+    const parentItem = currentItem.closest(`.${DefaultListCssClasses.item}`);
+
     /**
-     * Prevent editor.js behaviour
+     * Check that current item has any previous siblings to be merged with
      */
-    event.stopPropagation();
+    if (!previousItem && !parentItem) {
+      return;
+    }
 
     /**
-     * Prevent browser tab behaviour
+     * Make sure previousItem is an HTMLElement
      */
-    event.preventDefault();
+    if (previousItem && !isHtmlElement(previousItem)) {
+      return;
+    }
 
     /**
-     * Move item from current list to parent list
+     * Lets compute the item which will be merged with current item text
      */
-    this.unshiftItem();
-  }
+    let targetItem: Element | null;
 
+    /**
+     * If there is a previous item then we get a deepest item in its sublists
+     *
+     * Otherwise we will use the parent item
+     */
+    if (previousItem) {
+      const childrenOfPreviousItem = previousItem.querySelectorAll(`.${DefaultListCssClasses.item}`);
 
-  /**
-   * Decrease indentation of the current item
-   *
-   * @returns {void}
-   */
-  unshiftItem(): void {
-    const currentItem = this.currentItem;
-    if (!currentItem) {
-      return;
-    }
-    if (!currentItem.parentNode) {
-      return;
+      targetItem = childrenOfPreviousItem[childrenOfPreviousItem.length - 1] || previousItem;
+    } else {
+      targetItem = parentItem;
     }
-    if (!isHtmlElement(currentItem.parentNode)) {
+
+    /**
+     * Get current item content
+     */
+    const currentItemContent = currentItem.querySelector(`.${DefaultListCssClasses.itemContent}`)?.innerHTML ?? '';
+
+    /**
+     * Get the target item content element
+     */
+    if (!targetItem) {
       return;
     }
 
-    const parentItem = currentItem.parentNode.closest(`.${DefaultListCssClasses.item}`);
+    /**
+     * Get target item content element
+     */
+    const targetItemContent = targetItem.querySelector<HTMLElement>(`.${DefaultListCssClasses.itemContent}`);
```

Comment:
similar names, but `currentItemContent` stores a string, and `targetItemContent` stores HTMLElement. It is confusing.

Also, I'm suggesting to create a util for getting list item content